backup/src/main/java/io/camunda/zeebe/backup/management/InProgressBackupImpl.java

```diff
@@ -61,9 +125,19 @@ public ActorFuture<Void> findSegmentFiles() {
   }
 
   @Override
-  public ActorFuture<Void> save(final BackupStore store) {
-    // save backup
-    return concurrencyControl.createCompletedFuture();
+  public Backup createBackup() {
+    final String snapshotId;
+
+    if (hasSnapshot) {
+      snapshotId = reservedSnapshot.getId();
+    } else {
+      // To do : change snapshotId in BackupDescriptor to Optional
```

Comment:
:x: I think we need to adjust this now, https://github.com/camunda/zeebe/pull/10154 made `snapshotId`  optional

Reply:
It is not updated in `BackupDescriptor` yet.